### PR TITLE
Server only api

### PR DESCRIPTION
**Pull request by jaensen**
In this pull request I performed some major changes to the pathfinder code:
* Removed everything that's not directly required for the current production setup
* Improved the logging: Now shows the time, request-id, ip:port, function_name, graph-version, call-duration
* Added a call context to bundle all resources required to process a rpc call.
* Changed the way the different versions of the same EdgeDb are managed in worker threads

### CODE REVIEW

File: src/types/safe.rs [hunk 1]
```diff
@@ -19,28 +20,23 @@ impl Safe {
...
-                    amount - scaled_receiver_balance,
-                    self.balance(&self.token_address),
-                )
+                / one_hundred;
+
+            if max < receiver_balance {
+                return U256::from(0);
             }
+
+            let receiver_balance_scaled = receiver_balance * (one_hundred - U256::from(trust_percentage as u128)) / one_hundred;
```
Comment (llunaCreixent) [inline src/types/safe.rs]: Have you checked this issue?
We discovered that there is a potencial issue in the hub contract.
https://github.com/CirclesUBI/circles-contracts/issues/173

File: src/server.rs [hunk 2]
```diff
@@ -78,7 +78,7 @@ pub fn start_server(listen_at: &str, queue_size: usize, threads: u64) {
         thread::spawn(move || loop {
             let socket = rec.lock().unwrap().recv().unwrap();
             if let Err(e) = handle_connection(e.deref(), socket) {
-                println!("Error handling connection: {e}");
+                println!("Error handling connection: {}", e);
```
Comment (chriseth) [inline src/server.rs:79]: Wait what is this? I would have expected the opposite as that is the one recommended by `cargo clippy`.

**Comment by chriseth** [comment]
The "add capacity edges" commit looks good. Does it fix the issue?

**Comment by llunaCreixent** [comment]
> The "add capacity edges" commit looks good. Does it fix the issue?

It fixes partially the issue of the missing edges. Read more here: https://github.com/CirclesUBI/pathfinder2/pull/43

Furthermore, it seems the "token" added to the Edge is the token owner address and not the token address. Read more here: https://github.com/CirclesUBI/pathfinder2/pull/40#discussion_r1255678343

File: src/graph/augmenting_path.rs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    let mut queue = VecDeque::<(Node, (u64, U256))>::new();
+    queue.push_back((Node::Node(*source), (0, U256::default() - U256::from(1))));
+    while let Some((node, (depth, flow))) = queue.pop_front() {
+        if let Some(max) = max_distance {
+            // * 3 because we have three edges per trust connection (two intermediate nodes).
+            if depth >= max * 3 {
+                continue;
+            }
+        }
+        for (target, capacity) in adjacencies.outgoing_edges_sorted_by_capacity(&node) {
```
Comment (llunaCreixent) [inline src/graph/augmenting_path.rs:26]: Is the current implementation sorting only by capacity? I understand that we generating here a priority for the edges to be selected for the final path. Is here the place where we could also prioritize the edges that transfer tokens to their owners?

File: src/safe_db/db.rs [hunk 4]
```diff
@@ -68,4 +105,46 @@ impl DB {
...
+    /// If the receiver is an organization, the method simply returns the balance of the sender, i.e.,
+    /// the user can send all their tokens to an organization.
+    /// If the receiver is a regular user, the method calculates the transfer limit based on the balance
+    /// of the receiver and the trust percentage. The method scales down the receiver's balance based
+    /// on the trust percentage and compares it with the original balance to calculate the maximum
+    /// amount the user can send.
+    /// The maximum amount a user can send is the smaller of the user's balance and the difference between
+    /// the scaled receiver's balance and the balance calculated based on the trust percentage.
+    /// @returns how much of their own tokens a user can send to receiver.
+    fn trust_transfer_limit(&self
```
Comment (llunaCreixent) [inline src/safe_db/db.rs]: I'd improve the function by accepting the general case of sending any token, not only sender's own tokens, like in here: https://github.com/CirclesUBI/pathfinder2/pull/43/files#diff-4d5dbf803354caf6408bea02c0b734f850ad3aaa5d2ffea457510a0a2958c654

**Comment by JacqueGM** [comment]
@jaensen can you check the tests for the call_context... it's a required argument for the compute_flow but I dont know how to pass I tried different things but still failing could  you check my lasts commits and change update accordingly please

**Comment by benjaminbollen** [comment]
I tagged the deployed commit w [v0.1.0-server+202308](https://github.com/CirclesUBI/pathfinder2/tree/v0.1.0-server+202308) and pushed it as a new branch `current-prod` so that it can live next to `dev` until the two have been re-integrated.

see #55

**Comment by benjaminbollen** [comment]
I will close this PR as we temporarily will maintain `dev` and `current-prod`

**Pull request closed without merging**